### DISCUSSION TITLE
SpoutBreeze - The open-source webinars platform powered by BigBlueButton

### DISCUSSION THREAD

**Discussion by GhaziTriki** [spoutbreeze/spoutbreeze#31] [Annnouncements]
Dear everyone,

I am happy to announce that we will be providing to the community an open-source webinars platform.

Who are the "we"?
The answer is [Frictionless Solutions](https://frictionlesssolutions.com/); from USA and [RIADVICE](https://riadvice.tn/) from Tunisia. With this partnership **SpoutBreeze** was born.

The first architecture draft is available at the following link https://slideshare.net/riadvice/spoutbreeze-overview, and it contains everything you need to know so far.

It was also presented during ZKI workshop "On the further development of BigBlueButton" on February 26, 2021, more information can be found here https://www.cms.hu-berlin.de/en/zki-workshop-bbb?set_language=en

The documentation is available here https://docs.spoutbreeze.org/

The website https://spoutbreeze.org/ still have the "coming soon" but we will be putting more efforts on it once we have the first alpha version released.

For the development, we reached a level where most of the available components are compiling and working alone. The remaining is now linking them together to have a consistent platform.

All contributions to the projects are welcome, we just require to sign the Contributor Agreement https://spoutbreeze.org/SpoutBreezeContributorAgreement.pdf and that's all.

We will be sharing more information soon.

**Comment by GhaziTriki** [comment]
Hi Everyone,

We did not share any news since a while. This time we are back with some updates:
- We abandoned the idea of not recompiling the HTML5 client of BigBlueButton because it will make the JavaScript code for the `extender` project too complicated. From the team, @Mohith07 came with the following reflection `Even if we get access to the DOM with some workaround, we will need to do a lot of work to manage the state of the application and let react be aware of the current state.`. So we will maintain a fork of BigBlueButton and do our best not to touch any other component.
- We are now preparing the Jenkins server to compile and package the components. For now we will rely on zip archives then work on improving it and creating ubuntu Debian packages.
- @GhaziTriki will be presenting a demo of SpoutBreeze if possible during the `BigBlueButton World` event https://bigbluebutton.org/event-page/. Nothing is promised but we are doing our best to release a first version.
- We have almost finished the API entries for start and stopping the broadcast.
- We are looking for some help understanding why YouTube broacasting is not working and when broadcasting works audio is not audible in the stream.

Thank you for reading it, and looking forward to share more news very soon.

**Comment by lonesomewalker** [comment]
@GhaziTriki  if you need help with the RTMP part, feel free to contact...

**Reply by GhaziTriki** [reply]
Hi @lonesomewalker we definitely need some help but for ffmpeg inside selenoid. A small issue preventing audio from being broadcasted. Would you like to get in touch?

**Reply by lonesomewalker** [reply]
Sure :-)

**Reply by FreeScholar** [reply]
Is it dead?

**Reply by GhaziTriki** [reply]
@FreeScholar The project is not dead, just on-hold.

**Reply by samuelwei** [reply]
@GhaziTriki Any updates on this project?

**Reply by GhaziTriki** [reply]
Hello @SamuelWei. The project architecture has been completely revamped to be Kubernetes ready. It has a BigBlueButton plugin, can stream to twitch, exchange chat with Twitch IRC. I hope we can continue and improve it however a promising open-source without resources can hardly live.

**Reply by samuelwei** [reply]
Thanks for letting me know. I’ll present our work the BBB-Livestream-Server (https://github.com/THM-Health/BBB-Streaming-Server) on Wednesday at the BBB World, and I wanted to talk a minute about other projects in this field as an introduction.

**Reply by GhaziTriki** [reply]
Good to know @SamuelWei. I have checked "BBB-Streaming-Server" and "SpoutBreeze" comes with a complete different architecture. The idea behind it is to create a webinard platfrom that uses BigBlueButton under the hood, not committed only to stream a BigBlueButton session to any RTMP point, the project is taking a complete different direction.

**Reply by GhaziTriki** [reply]
@SamuelWei Best wishes for the BBB World talk by the way !
